addons/project/models/project.py

```diff
@@ -2271,25 +2271,16 @@ def _mail_track(self, tracked_fields, initial_values):
         if depends_changes and self.allow_task_dependencies and self.user_has_groups('project.group_project_task_dependencies'):
             parent_ids = self.dependent_ids
             if parent_ids:
-                fields_to_ids = self.env['ir.model.fields']._get_ids('project.task')
-                field_ids = [fields_to_ids.get(name) for name in depends_changes]
-                depends_tracking_value_ids = [
-                    tracking_values for tracking_values in tracking_value_ids
-                    if tracking_values[2]['field'] in field_ids
-                ]
-                subtype = self.env['ir.model.data']._xmlid_to_res_id('project.mt_task_dependency_change')
-                # We want to include the original subtype message coming from the child task
-                # for example when the stage changes the message in the chatter starts with 'Stage Changed'
-                child_subtype = self._track_subtype(dict((col_name, initial_values[col_name]) for col_name in changes))
-                child_subtype_info = child_subtype.description or child_subtype.name if child_subtype else False
-                # NOTE: the subtype does not have a description on purpose, otherwise the description would be put
-                #  at the end of the message instead of at the top, we use the name here
-                body = self.env['ir.qweb']._render('project.task_track_depending_tasks', {
-                    'child': self,
-                    'child_subtype': child_subtype_info,
-                })
+                old_stage = initial_values.get('stage_id')
                 for p in parent_ids:
-                    p.message_post(body=body, subtype_id=subtype, tracking_value_ids=depends_tracking_value_ids)
+                    blocking_tasks = p.depend_on_ids - self
+                    if all(task.is_closed and not task.is_blocked for task in blocking_tasks):
```

Comment:
`p.depend_on_ids - self` is literally `[record for record in p.depend_on_ids if not record in self]`.
So, to avoid looping twice, let's refactor it like that :
```suggestion
                    if all(task.is_closed and not task.is_blocked for task in p.depend_on_ids if not task in self):
```
But, as I said in the first review (you might have missed it), it is better to continue if the opposite condition, to gain an indent :
```suggestion
                    if any(not task.is_closed or task.is_blocked for task in p.depend_on_ids if not task in self):
                        continue
```